src/utils/lru-cache.js

```diff
@@ -16,49 +16,89 @@
 
 import {dev} from '../log';
 
-/** @typedef {{
-      payload: *,
-      access: number,
-    }} */
-export let Cacheable;
-
 /** @const {string} */
 const TAG = 'lru-cache';
 
-export class LRUCache {
-  /** @param {number} capacity */
+/**
+ * @template T
+ */
+export class LruCache {
+  /**
+   * @param {number} capacity
+   */
   constructor(capacity) {
-    /** @private {number} */
+    /** @private @const {number} */
     this.capacity_ = capacity;
 
-    /** @private {!Object<(number|string), !Cacheable>} */
+    /** @private {number} */
+    this.size_ = 0;
+
+    /**
+     * An incrementing counter to define the last access.
+     * @private {number}
+     */
+    this.access_ = 0;
```

Comment:
TIL that number overflow is not an issue.

Reply:
Yah, `2 ** 53` is a very large number. For reference, we can do 1,000 accesses every millisecond for 285 years before we hit it.

Reply:
I meant in JS in general, but +1 for napkin math!